n2v/methods/rmks.py

```diff
@@ -6,7 +6,521 @@
+            rho1inv = (1 / rho1)[:, None]
+
+            dvp_l = np.zeros_like(l_x)
+
+            # if not restricted:
+            #     dvp_l_b = np.zeros_like(l_x)
+    
+            # Loop over the inner set of blocks
+            for r_block in range(self.Vpot.nblocks()):
+                r_grid = self.Vpot.get_block(r_block)
+                r_w = np.array(r_grid.w())
+                r_x = np.array(r_grid.x())
+                r_y = np.array(r_grid.y())
+                r_z = np.array(r_grid.z())
+                r_npoints = r_w.shape[0]
+    
+                points_func.compute_points(r_grid)
+    
+                r_lpos = np.array(r_grid.functions_local_to_global())
+    
+                # Compute phi!
+                r_phi = np.array(points_func.basis_values()["PHI"])[:r_npoints, :r_lpos.shape[0]]
+    
+                # Build a local slice of D
+                if self.wfn.name() == "CIWavefunction":
+                    lD2 = D2[(r_lpos[:, None], r_lpos)]
+                    rho2 = contract('pm,mn,pn->p', r_phi, lD2, r_phi)
+    
+                    p, q, r, s = np.meshgrid(l_lpos, l_lpos, r_lpos, r_lpos, indexing="ij")
+                    Tap_temp = Tau_ijkl[p, q, r, s]
+    
+                    n_xc = contract("mnuv,pm,pn,qu,qv->pq", Tap_temp, l_phi, l_phi, r_phi, r_phi)
+                    n_xc *= rho1inv
+                    n_xc -= rho2
+                elif self.wfn.name() == "RHF":
+                    lD2 = self.wfn.Da().np[(l_lpos[:, None], r_lpos)]
+                    n_xc = - 2 * contract("mu,nv,pm,pn,qu,qv->pq", lD2, lD2, l_phi, l_phi, r_phi, r_phi)
+                    n_xc *= rho1inv
+
+                # Build the distnace matrix
+                R2 = (l_x[:, None] - r_x) ** 2
+                R2 += (l_y[:, None] - r_y) ** 2
+                R2 += (l_z[:, None] - r_z) ** 2
+                # R2 += 1e-34
+                if np.any(np.isclose(R2, 0.0, atol=atol)):
+                    # R2[np.isclose(R2, 0.0)] = np.min(R2[~np.isclose(R2, 0.0)])
+                    R2[np.isclose(R2, 0.0, atol=atol)] = np.inf
+                Rinv = 1 / np.sqrt(R2)
+    
+                # if restricted:
+                dvp_l += np.sum(n_xc * Rinv * r_w, axis=1)
+
+            # if restricted:
+            vxchole[w1_old:w1_old + l_npoints] += dvp_l
+            w1_old += l_npoints
+    
+        print("\n")
+        print("Totally %i grid points takes %.2fs with max %i points in a block."
+              % (vxchole.shape[0], time.time() - start_time, psi4.core.get_global_option("DFT_BLOCK_MAX_POINTS")))
+        assert w1_old == vxchole.shape[0], "Somehow the whole space is not fully integrated."
+        if blocks is None:
+            # if restricted:
+            self.vxc_hole_WF = vxchole
+            return self.vxc_hole_WF
+        else:
+            # if restricted:
+            return vxchole
+
+    def _average_local_orbital_energy(self, D, C, eig, grid_info=None):
+        """
+        (4)(6) in mRKS.
```

Comment:
When having functions that generate a specific function, it would be nice to have the formula's latex version. This is mainly because `sphinx,` the code that will automatically generate our documentation, has the ability to render latex. So that could be super helpful.